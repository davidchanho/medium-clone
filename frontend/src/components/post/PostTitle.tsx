import React from "react";
import { Card } from "react-bootstrap";
import { PostProps } from "../../types";

function PostTitle({ post }: PostProps) {
  return <Card.Title className="text-capitalize">{post.title}</Card.Title>;
}

export default PostTitle;
