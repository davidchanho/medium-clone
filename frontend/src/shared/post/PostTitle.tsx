import React from "react";
import { Card } from "react-bootstrap";
import { PostProps } from "../../store/posts/types";

function PostTitle({ post, className }: PostProps) {
  if (!post) {
    return null;
  }

  return (
    <Card.Title className={`text-capitalize mb-1 ${className}`}>
      {post.title}
    </Card.Title>
  );
}

export default PostTitle;
