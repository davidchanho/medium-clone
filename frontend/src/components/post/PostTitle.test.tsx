import React from "react";
import renderer from "react-test-renderer";
import { testPost } from "../../tests";
import PostTitle from "./PostTitle";

describe("PostTitle", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<PostTitle post={testPost} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
