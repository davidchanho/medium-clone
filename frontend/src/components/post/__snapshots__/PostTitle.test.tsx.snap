// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostTitle renders correctly 1`] = `
<div
  className="text-capitalize mb-1 card-title h5"
>
  test
</div>
`;
